recipe/session/sessionFunctions.go

```diff
@@ -61,51 +52,57 @@ func createNewSessionHelper(recipeImplHandshakeInfo *sessmodels.HandshakeInfo, c
 	if err != nil {
 		return sessmodels.CreateOrRefreshAPIResponse{}, err
 	}
+
 	return resp, nil
 }
 
 func getSessionHelper(recipeImplHandshakeInfo *sessmodels.HandshakeInfo, config sessmodels.TypeNormalisedInput, querier supertokens.Querier, parsedAccessToken ParsedJWTInfo, antiCsrfToken *string, doAntiCsrfCheck, containsCustomHeader bool) (sessmodels.GetSessionResponse, error) {
-	err := getHandshakeInfo(&recipeImplHandshakeInfo, config, querier, false)
-	if err != nil {
-		return sessmodels.GetSessionResponse{}, err
+	var accessTokenInfo *accessTokenInfoStruct = nil
+	var err error = nil
+	combinedJwks, jwksError := sessmodels.GetCombinedJWKS()
+	if jwksError != nil {
+		if !defaultErrors.As(jwksError, &errors.TryRefreshTokenError{}) {
+			return sessmodels.GetSessionResponse{}, jwksError
```

Comment:
`jwksError` can be a 500 because no core was found or if the JWKS endpoint in the core threw a 500.

`err` (later in the same function) can be safely assumed to always be try refresh token. At the time I simply clones the node logic but I can refactor this if needed